artiq/compiler/embedding.py

```diff
@@ -187,6 +187,11 @@ def _add(self, fragment):
         return source.Range(self.source_buffer, range_from, range_to,
                             expanded_from=self.expanded_from)
 
+    def _add_iterable(self, fragment):
+        if len(self.source) - self.source.rfind("\n") >= 2**16 - 2:
```

Comment:
> Sounds like this will break if you add an element that doesn't fit in 2 characters, no?

No, DILocation points on the beginning of the code. 
-2 is to fit the line break